app/src/main/java/com/github/libretube/util/HtmlParser.kt

```diff
@@ -0,0 +1,134 @@
+package com.github.libretube.util
+
+import android.text.Editable
+import android.text.Html
+import android.text.Spanned
+import androidx.core.text.HtmlCompat
+import org.xml.sax.Attributes
+import org.xml.sax.ContentHandler
+import org.xml.sax.Locator
+import org.xml.sax.SAXException
+import org.xml.sax.XMLReader
+
+class HtmlParser private constructor(private val handler: TagHandler) :
```

Comment:
Why using a private constructor here?

Reply:
To get what we need, we do need to override the link handling. This is the only way I find to be useful to handle timestamps because we could not find any callback from the default implementations.

Reply:
I am making it short.